context/app/static/js/pages/Dataset/utils.ts

```diff
@@ -1,11 +1,16 @@
 import { ProcessedDatasetInfo } from './hooks';
 
+const potentialDuplicateStates = ['error'];
+
 export function datasetSectionId(
-  dataset: Pick<ProcessedDatasetInfo, 'pipeline' | 'hubmap_id' | 'status'>,
+  dataset: Pick<ProcessedDatasetInfo, 'hubmap_id' | 'status'> & { pipeline?: string },
```

Comment:
Is there a new potential use case where a `pipeline` is not given, or is this for testing purposes only?

Reply:
I came across some processed dataset descendants in new/error states that did not have pipeline info on dev during the initial implementation of unified views, so I figured that it might be good to document this possibility in the type signature of this utility to make sure that isn't lost. We already had the `??` logic in place to fall back from the pipeline to the hubmap id if the pipeline info was missing.

Reply:
Got it!